lib/ansible/plugins/inventory/gitlab_runners.py

```diff
@@ -0,0 +1,130 @@
+            type: str
+            required: true
+            choices: gitlab_runners
+        server_url:
+            description: The URL of the Gitlab server, with protocol (i.e. http or https).
+            type: str
+            required: true
+            default: https://gitlab.com
+        api_token:
+            description: Gitlab token for logging in.
+            type: str
+            aliases:
+              - private_token
+              - access_token
+        filter:
+            description: filter runners from Gitlab API
+            type: str
+            choices: ['active', 'paused', 'online', 'specific', 'shared']
+        verbose_output:
+            description: Toggle to (not) include all available nodes metadata
+            type: bool
+            default: yes
+'''
+
+EXAMPLES = '''
+# gitlab_runners.yml
+plugin: gitlab_runners
+host: https://gitlab.com
+
+# Example using constructed features to create groups and set ansible_host
+plugin: gitlab_runners
+host: https://gitlab.com
+strict: False
+keyed_groups:
+  # add e.g. amd64 hosts to an arch_amd64 group
+  - prefix: arch
+    key: 'architecture'
+  # add e.g. linux hosts to an os_linux group
+  - prefix: os
+    key: 'platform'
+  # create a group per runner tag
+  # e.g. a runner tagged w/ "production" ends up in group "label_production"
+  # hint: labels containing special characters will be converted to safe names
+  - key: 'tag_list'
+    prefix: tag
+'''
+
+from ansible.errors import AnsibleError, AnsibleParserError
+from ansible.module_utils._text import to_native
+from ansible.plugins.inventory import BaseInventoryPlugin, Constructable
+
+try:
+    import gitlab
+    HAS_GITLAB = True
+except ImportError:
+    HAS_GITLAB = False
+
+
+class InventoryModule(BaseInventoryPlugin, Constructable):
+    ''' Host inventory parser for ansible using Gitlab API as source. '''
+
+    NAME = 'gitlab_runners'
+
+    def _populate(self):
+        self.gl = gitlab.Gitlab(self.get_option('server_url'), private_token=self.get_option('api_token'))
+        self.inventory.add_group('all')
+        try:
+            if self.get_option('filter'):
+                self.runners = self.gl.runners.all(scope=self.get_option('filter'))
+            else:
+                self.runners = self.gl.runners.all()
+            for self.runner in self.runners:
+                self.host = str(self.runner['id'])
+                self.ip_address = self.runner['ip_address']
+                self.host_attrs = vars(self.gl.runners.get(self.runner['id']))['_attrs']
+                self.inventory.add_host(self.host)
+                self.inventory.set_variable(self.host, 'ansible_host', self.ip_address)
+                if self.get_option('verbose_output', True):
+                    self.inventory.set_variable(self.host, 'gitlab_runner_attributes', self.host_attrs)
```

Comment:
If found this part weird with using `self` for temporary variables

Reply:
You mean `self.host`? Should i change it?